GWFish/modules/inspiral_corrections.py

```diff
@@ -0,0 +1,795 @@
+                + (chi_PN - 1)*(-0.06782917938621007 - 0.6689834970767117*eta + 3.403147966134083*eta2)\
+                + (chi_PN - 1)**2*(-0.05296577374411866 - 0.9923793203111362*eta + 4.820681208409587*eta2)\
+                + (chi_PN - 1)**3*(-0.006134139870393713 - 0.38429253308696365*eta + 1.7561754421985984*eta2)
+    
+            
+        # Conjunction frequencies
+        f3_amp = (np.abs(ff_RD + (ff_damp*gamma3*(np.sqrt(1-gamma2**2.) - 1)/gamma2)))
+        f2_amp = (f1_amp + f3_amp)/2.
+
+        
+        amp_MR = gamma1*(gamma3*ff_damp*ones)/((ff - ff_RD*ones)**2. + (gamma3*ff_damp*ones)**2)*np.exp(-gamma2*(ff - ff_RD*ones)/(gamma3*ff_damp*ones))
+    
+
+        amp_MR_f3, amp_MR_prime_f3 = phenomD_amp_MR(f3_amp, self.gw_params, ff_damp, ff_RD, gamma1, gamma2, gamma3)
+        amp_MR_f3 = float(amp_MR_f3)
+        amp_MR_prime_f3 = float(amp_MR_prime_f3)
+       
+        # Solve for delta coefficients (intermediate phase)
+        A = np.array([[1., f1_amp, f1_amp**2., f1_amp**3., f1_amp**4.],\
+                        [1., f2_amp, f2_amp**2., f2_amp**3., f2_amp**4.],\
+                        [1., f3_amp, f3_amp**2., f3_amp**3., f3_amp**4.],\
+                        [0., 1., 2.*f1_amp, 3.*f1_amp**2., 4.*f1_amp**3.],\
+                        [0., 1., 2.*f3_amp, 3.*f3_amp**2., 4.*f3_amp**3.]])
+        
+        b = np. array([amp_ins_f1, v2 , amp_MR_f3, amp_ins_prime_f1, amp_MR_prime_f3])
+        delta = np.linalg.solve(A, b)
+
+        
+        # Full intermediate amplitude
+        amp_int = (delta[0] + delta[1]*(ff) + delta[2]*(ff)**2. + delta[3]*(ff)**3. +  delta[4]*(ff)**4.)
+
+        ff1_amp = f1_amp*ones
+        ff3_amp = f3_amp*ones
+    
+        theta_minus1_amp = 0.5*(1*ones - step_function(ff,ff1_amp))
+        theta_minus2_amp = 0.5*(1*ones - step_function(ff,ff3_amp))
+    
+        theta_plus1_amp = 0.5*(1*ones + step_function(ff,ff1_amp))
+        theta_plus2_amp = 0.5*(1*ones + step_function(ff,ff3_amp))
+    
+        # Overall (2,2) mode factor and its derivative
+        A0 = 1./(np.pi**(2./3.))*(5./24.)**(0.5)*cst.c/r*Mc**(5./6.)*frequencyvector**(-7./6.) 
+
+        ####################### AMPLITUDE COMPONENTS ###########################
+        ###################### written continuosly in frequency ################
+
+        amp_ins = amp_ins*theta_minus1_amp*A0
+        amp_int = theta_plus1_amp*amp_int*theta_minus2_amp*A0
+        amp_MR = theta_plus2_amp*amp_MR*A0
+    
+        amp_tot = amp_ins + amp_int + amp_MR
+        
+        ############################### PROJECTIONS ############################
+        
+        hp = amp_tot*0.5*(1 + np.cos(iota)**2.)
+        hc = amp_tot*np.cos(iota)
+        polarizations = np.hstack((hp * phase, hc * 1.j * phase))
+
+        ############################### OUTPUT #################################
+
+        self._frequency_domain_strain = polarizations
+
+        ########################################################################

```

Comment:
This also copies over most of the code from TaylorF2 and PhenomD, we don't do that in Python :) 

Instead, we should break the standard code `waveforms.py` into more functions and attributes (self.something). Then, for PPE corrections, we should make child classes here in `inspired_corrections.py`, and PPE corrections should be added by overwriting specific attributes/functions in a short way, such that we don't copy the code over. Can you do this, @AlessandroAgapito23 ?

We can discuss it on the next Thursday's meeting from 2pm to 4 pm, and perhaps make a quick code session together on Zoom (CC @jacopok, @u-dupletsa , @janosch314  ).

Reply:
Hi all, I have worked on this type of modification in waveforms.py and inspiral_corrections.py.
I have also added a brief comment at the beginning of the script, in order to explain and resume all the new functions and attributes.
I was also thinking, upon Jacopo's suggestion, about some tests on the new waveforms. I'm going to work on them.